tensorflow/core/common_runtime/rendezvous_mgr.cc

```diff
@@ -65,10 +65,10 @@ void IntraProcessRendezvous::SameWorkerRecvDone(
     StatusCallback done) {
   // Do a quick copy (sharing the underlying buffer) if both tensors
   // are on host memory.
-  const bool src_host =
-      (send_args.alloc_attrs.on_host() || parsed.src.type == "CPU");
-  const bool dst_host =
-      (recv_args.alloc_attrs.on_host() || parsed.dst.type == "CPU");
+  const bool src_host = (send_args.alloc_attrs.on_host() ||
+                         parsed.src.type == "CPU" || parsed.src.type == "SYCL");
```

Comment:
My understanding was that the SYCL part of Eigen will take care of data management and TensorFlow will deal only with the host memory.
Unfortunately, I was proven wrong  by very simple python test that I created. 
To this end I am studying the codebase in order to implement missing functionality in the SYCL TensorFlow code. Which seems to be pointing to device specific allocator.
And yes you are right, SYCL has different address space for the data. SYCL manages data copies and dependencies internally via buffers and accessors and will maintain a CPU copy of the data when the buffer was initialised with a pointer.

> the tensor being used as input to the sycl device must remain in CPU memory.
Could you clarify exactly what you mean by that? Does that mean CUDA is deallocating memory once pushed to the Device memory?